tw-tasks-core/src/main/java/com/transferwise/tasks/stucktasks/TasksResumer.java

```diff
@@ -63,49 +75,51 @@ public void init() {
-                }
-
-                log.info("Stopping scheduled tasks executor.");
-                if (scheduledTaskHandleHolder.getValue() != null) {
-                  scheduledTaskHandleHolder.getValue().stop();
-                }
-
-                if (stuckTaskHandleHolder.getValue() != null) {
-                  stuckTaskHandleHolder.getValue().waitUntilStopped(Duration.ofMinutes(1));
-                }
-                log.info("Stuck tasks resumer stopped.");
-                if (scheduledTaskHandleHolder.getValue() != null) {
-                  scheduledTaskHandleHolder.getValue().waitUntilStopped(Duration.ofMinutes(1));
-                }
-                log.info("Scheduled tasks executor stopped.");
-              });
-        });
+    ILock lock = lockBuilderFactory.createBuilder(nodePath).build();
+    leaderSelector = new LeaderSelectorV2.Builder().setLock(lock).setExecutorService(executorService).setLeader(control -> {
+      ScheduledTaskExecutor scheduledTaskExecutor = executorServicesProvider.getGlobalScheduledTaskExecutor();
+      MutableObject<ScheduledTaskExecutor.TaskHandle> stuckTaskHandleHolder = new MutableObject<>();
+      MutableObject<ScheduledTaskExecutor.TaskHandle> scheduledTaskHandleHolder = new MutableObject<>();
+
+      control.workAsyncUntilShouldStop(
+          () -> {
+            stuckTaskHandleHolder.setValue(scheduledTaskExecutor.scheduleAtFixedInterval(() -> resumeStuckTasks(control),
+                tasksProperties.getStuckTasksPollingInterval(), tasksProperties.getStuckTasksPollingInterval()));
+
+            log.info("Started to resume stuck tasks after each " + tasksProperties.getStuckTasksPollingInterval() + " for '"
+                + tasksProperties.getGroupId() + "'.");
+
+            scheduledTaskHandleHolder.setValue(scheduledTaskExecutor.scheduleAtFixedInterval(() -> {
+              if (paused) {
+                return;
+              }
+              resumeWaitingTasks(control);
+            }, tasksProperties.getWaitingTasksPollingInterval(), tasksProperties.getWaitingTasksPollingInterval()));
+
+            log.info("Started to resume scheduled tasks after each " + tasksProperties.getWaitingTasksPollingInterval() + " for '"
+                + tasksProperties.getGroupId() + "'.");
+          },
+          () -> {
+            log.info("Stopping stuck tasks resumer for '" + tasksProperties.getGroupId() + "'.");
+            if (stuckTaskHandleHolder.getValue() != null) {
+              stuckTaskHandleHolder.getValue().stop();
+            }
+
+            log.info("Stopping scheduled tasks executor for '" + tasksProperties.getGroupId() + "'.");
+            if (scheduledTaskHandleHolder.getValue() != null) {
+              scheduledTaskHandleHolder.getValue().stop();
+            }
+
+            if (stuckTaskHandleHolder.getValue() != null) {
+              stuckTaskHandleHolder.getValue().waitUntilStopped(Duration.ofMinutes(1));
```

Comment:
Just to double check, we don't need to check the result of `waitUntilStopped` and log error otherwise, do we?